File: src/PlantBasedPizza.Api/modules/order-manager/PlantBasedPizza.OrderManager.Infrastructure/RecipeService.cs
using PlantBasedPizza.OrderManager.Core.Services;
using PlantBasedPizza.Recipes.DataTransfer;
using Recipe = PlantBasedPizza.OrderManager.Core.ViewModels.Recipe;

namespace PlantBasedPizza.OrderManager.Infrastructure
{
    public class RecipeService : IRecipeService
    {
        private readonly RecipeDataTransferService _recipes;

        public RecipeService(RecipeDataTransferService recipes)
        {
            _recipes = recipes;
        }

        public async Task<Recipe> GetRecipe(string recipeIdentifier)
        {
            var recipe = await this._recipes.GetRecipeAsync(recipeIdentifier);

            return new Recipe()
            {
                Price = recipe.Price,
                ItemName = recipe.Name,
            };
        }
    }
}

File: src/PlantBasedPizza.Kitchen.Node/src/adapters/recipeService.ts
import axios from "axios";
import { IRecipeService } from "../services/recipeService";
import { RecipeAdapter } from "./recipeAdapter";

export class RecipeService implements IRecipeService {
    async getRecipe(recipeIdentifier: string): Promise<RecipeAdapter | undefined> {
        try{
            if (process.env.INTEGRATION_TEST_RUN === "true"){
                return {
                    id: recipeIdentifier,
                    name: 'Test recipe',
                    category: 'Pizza',
                    price: 5,
                    ingredients:[{
                        name: 'test',
                        quantity: 1
                    }]
                }
            }
            
            const endpoint = `${process.env['RECIPE_API_ENDPOINT']}/recipes/${recipeIdentifier}`;

            console.log(`Querying '${endpoint}'`);
    
            const receipeResult = await axios.get<RecipeAdapter>(endpoint);
    
            console.log(`Result: ${receipeResult.status}`);
    
            return receipeResult.data;
        }
        catch (e) {
            console.log(e);

            return undefined;
        }
    }
}